consensus/src/decoupled_state_computer.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{error::StateSyncError, state_replication::StateComputer};
+use anyhow::Result;
+use consensus_types::{block::Block, executed_block::ExecutedBlock};
+use diem_crypto::{HashValue, hash::ACCUMULATOR_PLACEHOLDER_HASH};
+use diem_infallible::Mutex;
+use diem_logger::prelude::*;
+use diem_metrics::monitor;
+use diem_types::ledger_info::LedgerInfoWithSignatures;
+use execution_correctness::ExecutionCorrectness;
+use executor_types::{Error as ExecutionError, StateComputeResult};
+use fail::fail_point;
+use state_sync::client::StateSyncClient;
+use std::{boxed::Box, sync::Arc};
+
+/// Basic communication with the Execution module;
```

Comment:
we don't talk to execution in this implementation, we pass to some channels upon commit

Reply:
Resolved.